test/docker-compose-with-auth.yml

```diff
@@ -0,0 +1,7 @@
+version: "3"
+services:
+  docker-compose:
+    environment:
+      - REGISTRY_AUTH=htpasswd
+      - REGISTRY_AUTH_HTPASSWD_REALM=Sorealm
+      - REGISTRY_AUTH_HTPASSWD_PATH=/config/htpasswd
```

Comment:
See https://devilbox.readthedocs.io/en/latest/configuration-files/docker-compose-override-yml.html for how this works. Essentially you can give `docker-compose` multiple configuration files and it'll merge them together using a simple algorithm.